portal-wire-protocol.md

```diff
@@ -209,3 +210,42 @@ Upon *receiving* this message, the serving node **SHOULD** *initiate* a uTP stre
 
 A collection of test vectors for this specification can be found in the
 [Portal wire test vectors](./portal-wire-test-vectors.md) document.
+
+## Algorithms
+
+Here we define a collection of generic algorithms which can be applied to sub-protocol implementing the wire protocol.
+
+
+### Lookup
+
+We use the lookup algorithm described in section 2.3 of the Kademlia paper. 
+
+A "node lookup" is the execution of the algorithm to find the `k` closest nodes to some `node-id`. A "content lookup" is the execution of the algorithm to find the data for `content-id` or the `k` closest nodes to `content-id`.
+
+A `FindNode` request corresponds to a node lookup, and a `FindContent` request corresponds to a content lookup.
+
+The lookup algorithm is also used to identify nodes that should receive `Offer` messages to store some data.
```

Comment:
It is not actually the recursive mechanism (lookup) that is used here, but just the initial get closest neighbours, to the content, from own routing table. But I can address this in the neighborhood gossip section that I'll add.

Reply:
Although I did just realize that the `Finding Content` and Finding Nodes` sections briefly explain the POKE and Neighborhood gossip mechanisms. But I think it can still use a small update in another PR to specifically name/explain these concepts.